### PR TITLE
Fix False positive on Enum.__members__.items()

### LINKED ISSUE #4123: False positive on Enum.__members__.items()
### Steps to reproduce
1. Create an `Enum` subclass
2. Try to iterate over the Enum contents using `MyEnum.__members__.items()`
3. Lint the code

### Current behavior
observe `E1101: Function '__members__' has no 'items' member (no-member)`

### Expected behavior

Iterating over an Enum [per the Python docs](https://docs.python.org/3/library/enum.html#iteration) does not result in a linting error.

### pylint --version output
```
pylint 2.7.0
astroid 2.5
```

### PR DESCRIPTION
**Pull request by ikraduya**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [ ] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
Fix #4123 issue about "False positive on `Enum.__members__.items()`". I add a conditional at `_emit_no_member` function in typecheck.py.

I am curious if there are any other cases similar to this one.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |
|   | :sparkles: New feature |
|   | :hammer: Refactoring  |
|   | :scroll: Docs |

## Related Issue
Closes #4123

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you for this fix, we're going to cherry-pick it to 2.7.2, could you rebase and add a quick summary in `What's New in Pylint 2.7.2?` in `Changelog`, please ?

**Comment by ikraduya** [comment]
Sure, maybe I could also add `__members__.values` and `__members__.keys` as well?
As stated from this https://github.com/PyCQA/pylint/issues/4123#issuecomment-784518440

**Comment by Pierre-Sassoulas** [comment]
+1, closing two bugs with one MR is always nice :D

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
Thank you for the update @ikraduya , I'll merge as soon as pre-commit is fixed.

File: ChangeLog [hunk 1]
```diff
@@ -4,7 +4,27 @@ Pylint's ChangeLog
...
+..
+  Put new features here
 
+What's New in Pylint 2.7.2?
+===========================
+..
+  Put bug fixes that will be cherry-picked to latest major version here
+
+* Fix False Positive on `Enum.__members__.items()`, `Enum.__members__.values`, and `Enum.__members__.keys`
+
```
Comment (cdce8p) [inline ChangeLog:16]:
```suggestion

```
There are two trailing-whitespaces here which cause the pre-commit check to fail.
If you want, you could also run pre-commit locally with:
```bash
pip install pre-commit
pre-commit install
```
That will run the pre-commit checks on every commit.

**Review by cdce8p — CHANGES_REQUESTED** [review]
Could you rebase this branch onto the current `master` and force push it?
That would take care of some duplicate commits.

**Comment by Pierre-Sassoulas** [comment]
@ikraduya we have a sunday 28 at 8pm (tomorrow) for the release of `2.7.2` if we want this to be in the next debian :)

**Comment by Pierre-Sassoulas** [comment]
I squashed everything and used the fix by @cdce8p, it's all good. Thank you for the fix !

**Pull request merged**